src/main/java/com/limechain/network/protocol/blockannounce/BlockAnnounceEngine.java

```diff
@@ -85,20 +87,12 @@ private void handleBlockAnnounce(byte[] msg, PeerId peerId) {
         connectionManager.updatePeer(peerId, announce);
         warpSyncState.syncBlockAnnounce(announce);
         log.log(Level.FINE, "Received block announce for block #" + announce.getHeader().getBlockNumber() +
-                            " from " + peerId +
-                            " with hash:0x" + announce.getHeader().getHash() +
-                            " parentHash:" + announce.getHeader().getParentHash() +
-                            " stateRoot:" + announce.getHeader().getStateRoot());
+                " from " + peerId +
+                " with hash:0x" + announce.getHeader().getHash() +
+                " parentHash:" + announce.getHeader().getParentHash() +
+                " stateRoot:" + announce.getHeader().getStateRoot());
 
-        if (BlockState.getInstance().isInitialized()) {
-            try {
-                BlockState.getInstance().addBlock(new Block(announce.getHeader(), new BlockBody(new ArrayList<>())));
-            } catch (BlockNodeNotFoundException ignored) {
-                // Currently we ignore this exception, because our syncing strategy as full node is not implemented yet.
-                // And thus when we receive a block announce and try to add it in the BlockState we will get this
-                // exception because the parent block of the received one is not found in the BlockState.
-            }
-        }
+        BlockState.getInstance().addBlockToBlockTree(announce.getHeader());
```

Comment:
It would be good idea to add the check if the BlockState is initialized, because we do not use the BlockState when the node is started in light client mode.

Reply:
The check is inside the method and throws exception if the state is not initialized